src/math/src/number.rs

```diff
@@ -1,18 +1,545 @@
+    ///
+    /// # Error
+    /// Error::OutOfRange if the number is not in the range <-1, 1>
+    ///
+    /// ```
+    /// # use math::Number;
+    ///
+    /// # fn main() -> math::Result<()> {
+    ///     assert!(Number::from(-2).arcsin().is_err());
+    ///     assert!(Number::from(2).arcsin().is_err());
+    ///     let x = Number::random();
+    ///     let sin_x = x.sin()?;
+    ///
+    ///     assert_eq!(x, sin_x.arcsin()?);
+    /// #     Ok(())
+    /// # }
+    /// ```
+    pub fn arcsin(&self) -> Result<Self> {
+        todo!()
+    }
+
+    /// Computes the arccosine of a number. Return value is in radians in the range <0, pi>
+    ///
+    /// # Error
+    /// Error::OutOfRange if the number is not in the range <-1, 1>
+    ///
+    /// ```
+    /// # use math::Number;
+    ///
+    /// # fn main() -> math::Result<()> {
+    ///     assert!(Number::from(-2).arccos().is_err());
+    ///     assert!(Number::from(2).arccos().is_err());
+    ///     let x = Number::random();
+    ///     let cos_x = x.cos()?;
+    ///
+    ///     assert_eq!(x, cos_x.arccos()?);
+    ///
+    /// #     Ok(())
+    /// # }
+    /// ```
+    pub fn arccos(&self) -> Result<Self> {
+        todo!()
+    }
+
+    /// Computes the arctangent of a number. Return value is in radians in the range <-pi/2, pi/2>
+    ///
+    /// ```
+    /// # use math::Number;
+    ///
+    /// # fn main() -> math::Result<()> {
+    ///     let x = Number::random();
+    ///     let tan_x = x.tg()?;
+    ///
+    ///     assert_eq!(x, tan_x.arctg()?);
+    /// #     Ok(())
+    /// # }
+    /// ```
+    pub fn arctg(&self) -> Result<Self> {
+        todo!()
+    }
+
+    /// Computes the arccotangent of a number. Return value is in radians in the range <0, pi>
+    ///
+    /// ```
+    /// # use math::Number;
+    ///
+    /// # fn main() -> math::Result<()> {
+    ///     let x = Number::random();
+    ///     let cot_x = x.cotg()?;
+    ///
+    ///     assert_eq!(x, cot_x.arccotg()?);
+    ///
+    /// #     Ok(())
+    /// # }
+    /// ```
+    pub fn arccotg(&self) -> Result<Self> {
+        todo!()
+    }
+
+    /// Calculate combination number of the given `n` and `k`
+    ///
+    /// Since combination number is defined as `C(n, k)` mathematically
+    /// Neither `n` nor `k` is considered the center of the function
+    /// that's why this function does not taking `self` as parameter like other functions
+    ///
+    /// # Error
+    /// Error::FactorialNegative if either `n` or `k` is negative, because they will need to be factorialized
+    ///
+    /// ```
+    /// # use math::Number;
+    ///
+    /// # fn main() -> math::Result<Self> {
+    ///     assert!(Number::combination(-1, Number::random()).is_err());
+    ///     assert!(Number::combination(Number::random(), -1).is_err());
+    ///
+    ///     let n = Number::random();
+    ///
+    ///     // if k > n => C(n, k) == 0
+    ///     assert_eq!(Number::combination(4, 3)?, Number::ZERO);
```

Comment:
Swapped arguments for `Number::combination`.

Reply:
Oh I didn't notice it, thanks!